src/main/java/server/Client.java

```diff
@@ -20,6 +20,17 @@ public Client(Socket clientSocket) throws IOException {
 
     public PrintWriter getWriter() { return writer; }
 
+    public String chunkStream() {
+        char[] charBuffer = new char[100000];
```

Comment:
Ah so did you run into that BufferedReader problem where you don't know how much it has read already? I need to take a look at the http server spec this afternoon, but I'm assuming there aren't any example requests that are longer than 100,000 characters. Do you know about the content-length header?

Reply:
Yeah the way I had it before it wouldn't correctly read a request with a body, but you're right I could be tailoring the read length based on the content-length header of the request 👍

Reply:
Yep, that's what I was getting at. I don't actually know what happens if you try to read more characters than there are. Must be fine since I think that's happening with the way things are currently implemented. But I think you would run into trouble if the content length of the request was longer than x characters. Me asking about this is kind of left over from the old cob_spec suite of tests which required you to look at content-length. Not so sure it's necessary for this project.

Reply:
Yeah there wasn't anything in the `http-server-spec` suite that requires you to look at content-length headers, might be good to add that to the suite though (that suite in general could use some updating imo)